test/boost/sstable_directory_test.cc

```diff
@@ -111,21 +123,36 @@ highest_generation_seen(sharded<sstables::sstable_directory>& dir) {
 
 class wrapped_test_env {
```

Comment:
The wrapped_test_env was supposed to provide a way to extract sstables_manager from a test_env whatever the env is. Now it's turning into something bigger which deserves dedicated explanation